projects/storefrontlib/cms-components/misc/icon/icon.component.spec.ts

```diff
@@ -67,113 +56,412 @@ describe('IconComponent', () => {
+        expect(hostClassList).toContain('cartSymbol');
+        expect(hostClassList).not.toContain('flip-at-rtl');
+        expect(hostClassList).not.toContain('flip-at-ltr');
+      });
+
+      it('should generate non-sprited SVG', () => {
+        component.type = ICON_TYPE.INFO;
+        fixture.detectChanges();
+
+        expect(nativeDebugElement.childElementCount).toBe(1);
+
+        const svgElement = nativeDebugElement.children[0];
+        expect(svgElement.nodeName).toBe('svg');
+        expect(svgElement.attributes.length).toBe(0);
+        expect(svgElement.childElementCount).toBe(1);
+
+        const useElement = svgElement.children[0];
+        expect(useElement.nodeName).toBe('use');
+        expect(useElement.attributes.length).toBe(1);
+        expect(useElement.getAttribute('xlink:href')).toBe('#infoSymbol');
+        expect(useElement.childElementCount).toBe(0);
+      });
+
+      it('should generate sprited SVG', () => {
+        component.type = ICON_TYPE.CART;
+        fixture.detectChanges();
+
+        expect(nativeDebugElement.childElementCount).toBe(1);
+
+        const svgElement = nativeDebugElement.children[0];
+        expect(svgElement.nodeName).toBe('svg');
+        expect(svgElement.attributes.length).toBe(0);
+        expect(svgElement.childElementCount).toBe(1);
+
+        const useElement = svgElement.children[0];
+        expect(useElement.nodeName).toBe('use');
+        expect(useElement.attributes.length).toBe(1);
+        expect(useElement.getAttribute('xlink:href')).toBe(
+          './assets/sprite.svg#cartSymbol'
+        );
+        expect(useElement.childElementCount).toBe(0);
+      });
+
+      it('should generate a sprited SVG with a sanitized javascript: url', () => {
+        component.type = 'BAD_SVG';
+        fixture.detectChanges();
+
+        expect(nativeDebugElement.childElementCount).toBe(1);
+
+        const svgElement = nativeDebugElement.children[0];
+        expect(svgElement.nodeName).toBe('svg');
+        expect(svgElement.attributes.length).toBe(0);
+        expect(svgElement.childElementCount).toBe(1);
+
+        const useElement = svgElement.children[0];
+        expect(useElement.nodeName).toBe('use');
+        expect(useElement.attributes.length).toBe(1);
+        expect(useElement.getAttribute('xlink:href')).toBe(
+          'unsafe:javascript:alert(1)#<img src="." onerror="alert(5)">'
+        );
+        expect(useElement.childElementCount).toBe(0);
+      });
+    });
+
+    describe('TEXT icons', () => {
+      it('should contain a font symbol', () => {
+        component.type = 'HAPPY';
+        fixture.detectChanges();
+
+        expect(nativeDebugElement.textContent).toBe('😊');
+        expect(nativeDebugElement.childElementCount).toBe(0);
+
+        const hostClassList = nativeDebugElement.classList;
+        expect(hostClassList.length).toBe(2);
+        expect(hostClassList).toContain('cx-icon');
+        expect(hostClassList).toContain('😊');
```

Comment:
Not sure, if it makes sense to add the symbol of a TEXT icon to the CSS classes.  But afaict, the old implementation did it, too.